lib/Devel/Cover/Report/CoverallsJSON.pm

```diff
@@ -0,0 +1,175 @@
+package Devel::Cover::Report::CoverallsJSON;
+
+#
+# Generate Coveralls.io-compatible JSON report.
+#
+# Modified from http://cpansearch.perl.org/src/MIKIHOSHI/Devel-Cover-Report-Coveralls-0.11/lib/Devel/Cover/Report/Coveralls.pm
+#
+# Original code licensed under GPLv1:
+#
+#     https://raw.githubusercontent.com/kan/coveralls-perl/master/LICENSE
+#
+
+use strict;
+use warnings;
+use 5.008005;
+our $VERSION = "0.11";
+
+our $CONFIG_FILE  = '.coveralls.yml';
+our $API_ENDPOINT = 'https://coveralls.io/api/v1/jobs';
+our $SERVICE_NAME = 'coveralls-perl';
+
+use Devel::Cover::DB;
+use Devel::Cover::DB::IO::JSON;
+use HTTP::Tiny;
+use JSON::PP;
+use YAML;
+
+sub get_source
+{
+    my ( $file, $callback ) = @_;
+
+    my $source = '';
+    my @coverage;
+
+    open F, $file or warn( "Unable to open $file: $!\n" ), return;
+
+    while ( defined( my $l = <F> ) )
+    {
+        chomp $l;
+        my $n = $.;
+
+        $source .= "$l\n";
+        push @coverage, $callback->( $n );
+    }
+
+    close( F );
+
+    $file =~ s!^blib/!!;
+
+    return +{
+        name     => $file,
+        source   => $source,
+        coverage => \@coverage,
+    };
+}
+
+sub get_git_info
+{
+    my $git = {
+        head => {
+            id              => `git log -1 --pretty=format:'%H'`,
+            author_name     => `git log -1 --pretty=format:'%aN'`,
+            author_email    => `git log -1 --pretty=format:'%ae'`,
+            committer_name  => `git log -1 --pretty=format:'%cN'`,
+            committer_email => `git log -1 --pretty=format:'%ce'`,
+            message         => `git log -1 --pretty=format:'%s'`
+        },
+        remotes => [
+            map {
+                my ( $name, $url ) = split( " ", $_ );
+                +{ name => $name, url => $url }
+            } split( "\n", `git remote -v` )
+        ],
+    };
+    my ( $branch, ) = grep { /^\* / } split( "\n", `git branch` );
+    $branch =~ s/^\* //;
+    $git->{ branch } = $branch;
+
+    return $git;
+}
+
+sub get_config
+{
+    my $config = {};
+    if ( -f $CONFIG_FILE )
+    {
+        $config = YAML::LoadFile( $CONFIG_FILE );
+    }
+
+    my $json = {};
+
+    # $json->{repo_token} = $config->{repo_token} if $config->{repo_token};
+    # $json->{repo_token} = $ENV{COVERALLS_REPO_TOKEN} if $ENV{COVERALLS_REPO_TOKEN};
+
+    my $is_travis;
```

Comment:
if we are using this file, should we simplify it to our use case? for example, get rid of comments, get rid of this branching logic for different services.